### PR TITLE
HUD - Bars for Hunger/Stamina

### PR DESCRIPTION
**Pull request by Jacqueuela**
UI element of health + stamina bar done
hud stuff
Hunger/Stamina bars update
Renamed to have consistent naming
Commenting

### CODE REVIEW

File: Live/Classes/HUD/HUDBar.h [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+private:
+    // Components of a HUDBar
+    cocos2d::Label* m_label;
+    cocos2d::ProgressTimer* m_bar;
+
+    // Stored properties
+    std::string m_title;
+    double m_maxValue;
+    double m_currValue;
```
Comment (sadmansk) [inline Live/Classes/HUD/HUDBar.h]: `m_curr_value` sorry that's the C++ way of naming variables.

File: Live/Classes/HUD/HUDBar.h [hunk 2]
```diff
@@ -0,0 +1,36 @@
+#ifndef _HUD_BAR_
+#define _HUD_BAR_
```
Comment (sadmansk) [inline Live/Classes/HUD/HUDBar.h]: don't use header guards, use `#pragma once` instead, we are switching to that in other files too, slowly.

File: Live/Classes/HUD/HUDBar.h [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+#include "cocos2d.h"
+
+class HUDBar : public cocos2d::Layer {
+public:
+    HUDBar(const std::string& _title, double max, int x, int y);
```
Comment (sadmansk) [inline Live/Classes/HUD/HUDBar.h:7]: never prefix a variable name with `_`, those are reserved for the compiler.

File: Live/Classes/HUD/HUDBar.cpp [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    // update view
+    m_bar->setPercentage(m_currValue / m_maxValue * 100.0);
+    m_label->setString(m_title);
+}
+
+void HUDBar::updateCurr(double curr) { m_currValue = curr; }
+
+void HUDBar::updateMax(double max) { m_maxValue = max; }
+
+void HUDBar::setTitle(const std::string& _title) { m_title = _title; }
```
Comment (sadmansk) [inline Live/Classes/HUD/HUDBar.cpp]: again, remove the _.

File: Live/Classes/HUD/HUDBar.cpp [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+USING_NS_CC;
+
+HUDBar::HUDBar(const std::string& _title, double max, int x, int y) {
+    // Set properties
+    HUDBar::m_title = _title;
+    HUDBar::m_maxValue = max;
+    HUDBar::m_currValue = max;
+
+    // Create, customize, and add label
+    m_label = Label::createWithSystemFont(m_title.c_str(), "Ubuntu", 12);
```
Comment (sadmansk) [inline Live/Classes/HUD/HUDBar.cpp]: are there other system fonts we can use? idk if ubuntu would be available on all systems.

File: Live/Classes/HUD/HUD.h [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+#define _HUD_HPP_
+
+#include "cocos2d.h"
+#include "HUDBar.h"
+#include "Player.h"
+
+class HUD : public cocos2d::Layer {
+public:
+    // HUD stores a reference to game objects that have the potential to update the HUD
+    HUD(Player* player);
```
Comment (sadmansk) [inline Live/Classes/HUD/HUD.h]: does HUD have any control over the player? like can it change player's attributes? If not, store as a const.

**Review by sadmansk — CHANGES_REQUESTED** [review]
Also a general comment, we follow `variable_name` convention for naming variables, `functionNames` use camel case, variables use snake case.

File: Live/Classes/HUD/HUDBar.cpp (hunk 5 above)
Comment (agentflare) [reply Live/Classes/HUD/HUDBar.cpp]: Looking in to. Loading other fonts had errors.

**Review by sadmansk — APPROVED** [review]

**Pull request merged**